request.py

```diff
@@ -0,0 +1,127 @@
+import logging
+from datetime import datetime
+from sqlalchemy import exc
+import sqlalchemy
+from flask import jsonify, request, make_response, g
+from init import app
+import function
+# import utils
+from models import User
+from flask_httpauth import HTTPBasicAuth
+auth = HTTPBasicAuth()
+
+
+@app.route("/")
+def index():
+    return "Welcome to Hangman!"
+
+
+@app.route("/User", methods=['POST'])
+def create_user():
+    user_info = request.json
+    if request.method == 'POST':
+        try:
+            username = user_info["username"]
+            password = user_info["password"]
+            if User.query.filter_by(username=username).first() is not None:
+                return make_response(jsonify(response="Welcome " + username + ". "
+                                                      + "Please press 1 to login with your password as an existing user."), 402)
+            if username is None or password is None:
+                return make_response(jsonify(response="Please enter a valid username or password"), 405)
+
+            function.create_user(username=username, password=password)  # Don't know yet whether it is json.get or json
+            return make_response(jsonify(response='OK'), 201)
+        except sqlalchemy.exc.InvalidRequestError:
+            return make_response(jsonify(error='Bad request'), 400)
+
+
+@auth.verify_password
+def verify_password(username, password):
+    user = function.get_user(username)
+    if not user or not user.verify_password(password):
+        return False
+    g.user = user
+    return True
+
+
+# Scared I didn't use a get request. Hope this works.
+@app.route('/User/<username>')
+@auth.login_required
+def get_resource():
+    return jsonify({'data': 'Hello, %s!' % g.user.username})
+
+
+@app.route('/User/<username>/Game', methods=['POST', 'GET'])
+def create_game(username):
```

Comment:
It is preferable to route different endpoints (url + vb) to separate methods instead of the same method for multiple. Let flask do the heavy work for you.